code/online_challenges/src/leetcode/single_number/single_number.cpp

```diff
@@ -0,0 +1,34 @@
+/*
+Given a non-empty array of integers, every element appears twice except for one. Find that single one.
+
+Note:
+
+Your algorithm should have a linear runtime complexity. Could you implement it without using extra memory?
+
+Example 1:
+
+Input: [2,2,1]
+Output: 1
+
+Example 2:
+
+Input: [4,1,2,1,2]
+Output: 4
+*/
+
+
+
+class Solution 
+{
+    public:
+        int singleNumber(vector<int>& nums) 
```

Comment:
Missing #includes. Also should be `std::`.

Reply:
All problems in leetcode are functional problems and we just have to create the function corresponding to the question.